torch/csrc/jit/frontend/schema_type_parser.cpp

```diff
@@ -165,18 +165,35 @@ TypePtr SchemaTypeParser::parseRefinedTensor() {
         c10::nullopt);
   } else {
     std::vector<int64_t> dims;
+    bool seen_strides = false;
+    std::vector<int64_t> strides;
     parseList(TK_NOTHING, ',', ')', [&] {
       const std::string& num = L.expect(TK_NUMBER).text();
       std::string::size_type num_len;
       size_t dim = c10::stoi(num, &num_len);
-      AT_ASSERTM(
-          num_len == num.size(),
-          "Bad tensor dimension size. Strides not yet supported in parsing",
-          num);
       dims.push_back(dim);
+      if (L.cur().kind == ':') {
```

Comment:
what happens if some of the dimensions are missing the stride information? 
can we enforce that we either have the stride information for all dimensions or we don't?

Reply:
Initially I thought that it could be a user error, but thinking about it more I decided to detect and report it as you suggested. I've updated the PR accordingly!